### PR TITLE
Upgrated libray to angualr v17, and imported dynaic form library

### PR DESCRIPTION
**Pull request by syedameenabrar**

### CODE REVIEW

File: frontend-auth-module/projects/sl-auth-app/src/app/components/contact/contact.component.html
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-app/src/app/components/contact/contact.component.html:1]: Why are we pushing such components?

File: frontend-auth-module/projects/sl-auth-app/src/app/components/home/home.component.html
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-app/src/app/components/home/home.component.html:1]: same as contact component, if unused then do remove such files

File: frontend-auth-module/projects/sl-auth-app/src/app/components/observation/observation.component.html
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-app/src/app/components/observation/observation.component.html:1]: same as home, contact feedback

File: frontend-auth-module/projects/sl-auth-app/src/app/components/solution/solution.component.html
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-app/src/app/components/solution/solution.component.html:1]: same issue as observation, home, contact etc

File: frontend-auth-module/projects/sl-auth-app/src/main.ts [hunk 1]
```diff
@@ -1,12 +1,7 @@
-import { enableProdMode } from '@angular/core';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 
 import { AppModule } from './app/app.module';
-import { environment } from './environments/environment';
 
-if (environment.production) {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-app/src/main.ts:7]: why this has been removed?

File: frontend-auth-module/projects/sl-auth-app/src/polyfills.ts [hunk 2]
```diff
@@ -50,4 +50,4 @@ import 'zone.js';  // Included with Angular CLI.
 
 /***************************************************************************************************
  * APPLICATION IMPORTS
- */
+ */
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-app/src/polyfills.ts]: Such changes should not be there

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.css [hunk 3]
```diff
@@ -0,0 +1,116 @@
+.login-container {
+  width: 80%; 
+  max-width: 500px; 
+  padding: 20px;
+  background-color: #fff;
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.css]: why is the color hardcoded here?

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  
+
+  submitData() {
+    this.baseApiService
+      .post(
+        this.configData?.baseUrl,this.configData?.loginApiPath,this.formLib?.myForm.value)
+      .subscribe((res: any) => {
+        if (res?.message == "User logged in successfully.") {
+          // alert(res?.message);
+          localStorage.setItem('accToken',res?.result?.access_token);
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts]: Can you think of a better way to set data in localstorage in a more declarative way? maybe using a loop to store each item dynamically?

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.css [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+.back-arrow mat-icon {
+  font-size: 24px; 
+}
+
+.login-container {
+  width: 100%; 
+  max-width: 600px; 
+  padding: 20px;
+  background-color: #fff;
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.css]: hardcoded color

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.html
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.html:1]: Please format html, the spacing is such that its hard to review

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    }
+  }
+
+  otpInputConfig: NgxOtpInputConfig = {
+    otpLength: 6,
+    autofocus: true,
+    // classList: {
+    //   inputBox: "my-super-box-class",
+    //   input: "my-super-class",
+    //   inputFilled: "my-super-filled-class",
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: please remove commented code

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+  handleFillEvent(value: string): void {
+    this.otp = value;
+  }
+
+  navigateBack() {
+    this.location.back();
+  }
+
+  generateOTP() {
+    console.log(this.regFormData)
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: remove console logs

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+    let payload = {
+      email: this.regFormData?.email,
+      password: this.regFormData?.password,
+      name: this.regFormData?.name,
+    }
+
+    let selectedApiPath: string;
+    if (this.regFormData.fromPage === "signup") {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: can this logic be altered maybe using ternary operator?

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+        this.startTimer();
+      },
+        (err: any) => {
+          alert(err?.error?.message);
+        }
+      );
+  }
+
+
+  verifyOTP() {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: generateOTP and verifyOTP have similar lines of code, generalize the functions into a single function and write conditions specific to generate and verify only where its applicable

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/signup/signup.component.ts
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/signup/signup.component.ts:1]: there are a lot of similarities between singup and reset password components, see how we can create a generalized component and use

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/guard/auth.guard.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+})
+export class AuthGuard implements CanActivate {
+
+  constructor(private router: Router) {}
+
+  canActivate(route: any, state: RouterStateSnapshot): boolean | UrlTree | Observable<boolean | UrlTree> | Promise<boolean | UrlTree> {
+    
+    const isAuthenticated = localStorage.getItem('name');
+    if (isAuthenticated) {
+      if (state?.url === '/login' || state?.url === '/signup' || state?.url === '/landing' || state?.url === '/otp' || state?.url === '/reset-password') {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/services/guard/auth.guard.ts]: can you include method of this condition check and see how the whole logic of this guard can be simplified

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/loader/loader.service.ts
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/services/loader/loader.service.ts:1]: there's no logic here, please remove such services

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/state/state.service.ts
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/services/state/state.service.ts:1]: what's the use of this services and how and where are using this?

File: frontend-auth-module/projects/sl-auth-lib/src/lib/sl-auth-lib.component.ts
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/sl-auth-lib.component.ts:1]: can you help me understand this file? what's the use of this?

File: frontend-auth-module/projects/sl-auth-lib/src/lib/sl-auth-lib.service.ts
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/sl-auth-lib.service.ts:1]: Remove unused services

File: frontend-auth-module/projects/sl-auth-lib/tsconfig.lib.prod.json
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/tsconfig.lib.prod.json:1]: why this change?

**Review by 5Amogh — CHANGES_REQUESTED** [review]

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/state/state.service.ts
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/services/state/state.service.ts:1]: We are using it while "registering" and "reset-password" to generate otp, and to send input form data to otp page.

File: frontend-auth-module/projects/sl-auth-lib/src/lib/sl-auth-lib.component.ts
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/sl-auth-lib.component.ts:1]: This is the entry point for the library, where <router-outlet> is added

File: frontend-auth-module/projects/sl-auth-lib/src/lib/sl-auth-lib.service.ts
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/sl-auth-lib.service.ts:1]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/loader/loader.service.ts
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/services/loader/loader.service.ts:1]: Done

File: frontend-auth-module/projects/sl-auth-app/src/main.ts (hunk 1 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-app/src/main.ts:7]: i was not using env files so, i removed it.

I have added it again.

File: frontend-auth-module/projects/sl-auth-app/src/app/components/solution/solution.component.html
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-app/src/app/components/solution/solution.component.html:1]: done

File: frontend-auth-module/projects/sl-auth-app/src/app/components/observation/observation.component.html
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-app/src/app/components/observation/observation.component.html:1]: Done

File: frontend-auth-module/projects/sl-auth-app/src/app/components/home/home.component.html
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-app/src/app/components/home/home.component.html:1]: I need home component in sl-auth-app, for testing purpose

File: frontend-auth-module/projects/sl-auth-app/src/app/components/contact/contact.component.html
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-app/src/app/components/contact/contact.component.html:1]: done

File: frontend-auth-module/projects/sl-auth-app/src/polyfills.ts (hunk 2 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-app/src/polyfills.ts]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.css (hunk 3 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.css]: I have made it dynamic and changed "px" to "rem"

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts (hunk 4 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.css (hunk 5 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.css]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.html
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.html:1]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts (hunk 6 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts (hunk 8 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts (hunk 9 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/guard/auth.guard.ts (hunk 10 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/services/guard/auth.guard.ts]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/guard/auth.guard.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+  constructor(private router: Router) {}
+
+  canActivate(route: any, state: RouterStateSnapshot): boolean | UrlTree | Observable<boolean | UrlTree> | Promise<boolean | UrlTree> {
+    
+    const isAuthenticated = !!localStorage.getItem('name');
+    const publicRoutes = ['/login', '/signup', '/landing', '/otp', '/reset-password'];
+
+    if (isAuthenticated) {
+      if (publicRoutes.includes(state.url)) {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/services/guard/auth.guard.ts]: this logic can still be refined, Please check and make the changes

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+
+  handleFillEvent(value: string): void {
+    this.otp = value;
+  }
+
+  navigateBack() {
+    this.location.back();
+  }
+
+  processOTP(action: 'generate' | 'verify') {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts:78]: Good however this logic can still needs to be refined as there' scope for generalised due to overlapping logic

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+    }
+  }
+  
+  submitData() {
+    this.baseApiService
+      .post(
+        this.configData?.baseUrl,this.configData?.loginApiPath,this.formLib?.myForm.value)
+      .subscribe((res: any) => {
+        if (res?.result) {
+
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts]: see how this can be refined more, can those values be stored using a single for each instead of two

File: frontend-auth-module/projects/sl-auth-app/src/app/app-routing.module.ts [hunk 14]
```diff
@@ -1,10 +1,13 @@
...
-const routes: Routes = [];
+import { HomeComponent } from './components/home/home.component';
+import { AuthGuard } from 'projects/sl-auth-lib/src/public-api';
+const routes: Routes = [
+  {path:'home', component:HomeComponent, canActivate: [AuthGuard] }
+];
 
 @NgModule({
-  imports: [RouterModule.forRoot(routes)],
+  imports: [RouterModule.forChild(routes)],
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-app/src/app/app-routing.module.ts:10]: Why was forChild added instead of forRoot? what advantage is it giving us, can you please explain?

**Review by 5Amogh — CHANGES_REQUESTED** [review]

File: frontend-auth-module/projects/sl-auth-app/src/app/app-routing.module.ts (hunk 14 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-app/src/app/app-routing.module.ts:10]: To use this library in parent application, we can't use "forRoot" in auth-library it will create a conflicts and throw error.

So we need to make it as "forChild", to use it without any routing errors in parent application.

File: frontend-auth-module/projects/sl-auth-app/src/styles.scss [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+  color: black;
+  font-weight: 500;
+}
+
+.mdc-text-field--filled:not(.mdc-text-field--disabled) .mdc-line-ripple::before {
+  border-bottom-color: black !important;
+}
+
+
+
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-app/src/styles.scss]: remove this long space please

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/endpoint/endpoint.service.ts [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+  public configData: any;
+
+  constructor(@Inject(LIBRARY_CONFIG) public config: any, private http: HttpClient) { 
+    this.config.subscribe((data: any) => {
+      this.configData = data;
+    });
+  }
+
+  async getEndpoint(): Promise<any> {
+    return new Promise((resolve, reject) => {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/services/endpoint/endpoint.service.ts]: We can use an observable here instead of promise, why did we use promises here?

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/guard/auth.guard.ts [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  canActivate(
+    route: ActivatedRouteSnapshot, 
+    state: RouterStateSnapshot
+  ): boolean | UrlTree | Observable<boolean | UrlTree> | Promise<boolean | UrlTree> {
+
+    const isAuthenticated = !!localStorage.getItem('name');
+
+    if (isAuthenticated) {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/services/guard/auth.guard.ts]: this logic can be improved, please make it more readable

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/singup-reset-pass/singup-reset-pass.component.html [hunk 18]
```diff
@@ -0,0 +1,18 @@
+<div class="header">
+    <a (click)="navigateBack()" class="back-arrow">
+        <mat-icon>arrow_back</mat-icon>
+    </a>
+</div>
+<div class="login-container">
+    <h1>{{ mode === 'signup' ? 'Signup to' +' '+ configData?.projectName : 'Reset password' }}</h1>
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/singup-reset-pass/singup-reset-pass.component.html]: we can do this but see if we can avoid writing logic in this way

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/singup-reset-pass/singup-reset-pass.component.ts [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+    this.baseApiService = inject(ApiBaseService);
+    this.endPointService = inject(EndpointService);
+    this.router = inject(Router);
+    this.location = inject(Location);
+  }
+
+  ngOnInit() {
+    this.mode = this.route.snapshot.data['mode'];
+    if (this.mode === 'reset') {
+      this.formJson.controls = this.formJson.controls.filter((control: any) => control.name !== 'name');
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/singup-reset-pass/singup-reset-pass.component.ts:103]: what are we doing in this logic when the mode reset?

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts [hunk 20]
```diff
@@ -0,0 +1,127 @@
...
+      ? action === 'generate' ? "otpGenerationApiPathForRegistration" : "signUpApiPath"
+      : isReset
+        ? action === 'generate' ? "otpGenerationApiPathForResetPassword" : "resetPasswordApiPath"
+        : "";
+  
+    const redirectionPath = isSignup
+      ? this.configData?.redirectRouteOnLoginSuccess
+      : isReset
+        ? action === 'generate' ? undefined : "/login"
+        : undefined;
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts:113]: please set the final value as null instead of undefined

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts [hunk 21]
```diff
@@ -0,0 +1,127 @@
...
+      otp: action === 'verify' ? this.otp : undefined
+    };
+  
+    if (isReset && action === 'generate') {
+      delete payload.name;
+    }
+  
+    const selectedApiPath = isSignup
+      ? action === 'generate' ? "otpGenerationApiPathForRegistration" : "signUpApiPath"
+      : isReset
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: this condition check can be improved, pls check

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts [hunk 22]
```diff
@@ -0,0 +1,127 @@
...
+        (res: any) => {
+          alert(res?.message);
+          if (action === 'verify' && redirectionPath) {
+            this.router.navigateByUrl(redirectionPath);
+          } else if (action === 'generate') {
+            this.otpInput = true;
+            this.startTimer();
+          }
+  
+          const formContainer = document.querySelector('.login-container') as HTMLElement;
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: instead of querying dom directly can we leverage renderer2 for the same?

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts [hunk 23]
```diff
@@ -0,0 +1,127 @@
...
+    this.baseApiService
+      .post(this.configData?.baseUrl, this.configData?.[selectedApiPath], payload)
+      .subscribe(
+        (res: any) => {
+          alert(res?.message);
+          if (action === 'verify' && redirectionPath) {
+            this.router.navigateByUrl(redirectionPath);
+          } else if (action === 'generate') {
+            this.otpInput = true;
+            this.startTimer();
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts:130]: what if both if & else if condition fails? we need to handle that as well

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts [hunk 24]
```diff
@@ -0,0 +1,114 @@
...
+              localStorage.setItem(key, String(value));
+            });
+  
+            this.router.navigateByUrl(this.configData?.redirectRouteOnLoginSuccess);
+          } else {
+            alert(res?.message);
+          }
+        },
+        (err: any) => {
+          alert(err?.error?.message);
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts:109]: what will be the message here that user is going to see? will it be user friendly, have we tested this scenario?

**Review by 5Amogh — CHANGES_REQUESTED** [review]

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/singup-reset-pass/singup-reset-pass.component.ts (hunk 19 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/singup-reset-pass/singup-reset-pass.component.ts:103]: For reset there is no form control input  as "name", so removing it.

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts (hunk 20 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts:113]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts (hunk 24 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts:109]: I have changed the code and added a default message

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts (hunk 23 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts:130]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts (hunk 21 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.css [hunk 25]
```diff
@@ -0,0 +1,142 @@
...
+  margin: 3rem 0;
+}
+
+.or-section p span {
+  background: var(--color-light);
+  padding: 0 0.625rem;
+  font-weight: 500;
+}
+
+.haveAccountAlready {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.css]: please follow a consistent naming convetion ex:login-container follows kebab-case where haveAccountAlready follows camelCase

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.html [hunk 26]
```diff
@@ -0,0 +1,78 @@
...
+    <p><span>OR</span></p>
+  </section>
+  <h5 class="haveAccountAlready">Have an account already?</h5>
+  <button type="button" routerLink="/login">
+    <mat-icon>email</mat-icon><span>Login</span>
+  </button>
+</div>
+
+
+
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.html]: Pls remove space issues & unused code

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.ts [hunk 27]
```diff
@@ -0,0 +1,31 @@
...
+
+  constructor() { 
+    this.endPointService = inject(EndpointService);
+  }
+
+  ngOnInit() {
+    this.fetchConfigData();
+  }
+
+  async fetchConfigData() {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.ts]: please async ass we're not using await in this block

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts [hunk 28]
```diff
@@ -0,0 +1,122 @@
...
+        }
+      );
+  }
+  
+
+  navigateBack() {
+    if (window.history.length < 1) {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts:116]: this condition seems to be incorrect as there'll always be a history (including current route) so it'll never be less than one so it'll always route to `/landing` instead you might want to invert the condition when history lengh is more than one instead of less than one

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.css [hunk 29]
```diff
@@ -0,0 +1,214 @@
...
+
+.forgotPass{
+  color: var(--color-secondary);
+  font-weight: 500;
+  font-size: 1rem;
+  margin-bottom: 7%;
+  cursor: pointer;
+}
+
+.troubleInSingup{
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.css]: casing issue as previously mentioned

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts [hunk 30]
```diff
@@ -0,0 +1,150 @@
...
+  }
+
+  ngOnInit() {
+    this.startTimer();
+    this.regFormData = this.stateService.getData();
+    this.fetchConfigData();
+    this.formContainer = this.renderer.selectRootElement('.login-container', true);
+  }
+
+  async fetchConfigData() {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: remove async if w're not using await acorss the codebase

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/auth.service.ts [hunk 31]
```diff
@@ -0,0 +1,76 @@
...
+        payload
+      ).pipe(
+        catchError((error) => {
+          alert(error?.error?.message || 'An error occurred during logout');
+          throw error
+        })
+      )
+      .subscribe(
+        (res: any) => {
+          if (res?.responseCode === "OK") {
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-lib/src/lib/services/auth.service.ts:63]: instead of responseCode please see if we can use status or result? or is this better approach in this context?

File: frontend-auth-module/projects/sl-auth-app/src/app/components/home/home.component.html [hunk 32]
```diff
@@ -0,0 +1,2 @@
+<p>home works!</p>
```
Comment (5Amogh) [inline frontend-auth-module/projects/sl-auth-app/src/app/components/home/home.component.html]: `<p>home works!</p>`, pls remove such temp codes

**Review by 5Amogh — CHANGES_REQUESTED** [review]

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.css (hunk 25 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.css]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.html (hunk 26 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.html]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.ts (hunk 27 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/landing/landing.component.ts]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.css (hunk 29 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.css]: Done

File: frontend-auth-module/projects/sl-auth-app/src/app/components/home/home.component.html (hunk 32 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-app/src/app/components/home/home.component.html]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/services/auth.service.ts (hunk 31 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/services/auth.service.ts:63]: Screenshot from 2024-06-20 12-37-57

Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/services/auth.service.ts:63]: There is no other value to check the status in response

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts (hunk 30 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts]: Done

File: frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts (hunk 28 above)
Comment (syedameenabrar) [reply frontend-auth-module/projects/sl-auth-lib/src/lib/components/login/login.component.ts:116]: I tried that way but it did not work, Scenario is that when we land directly to login page and try to click back button it did not navigate to any page, because there was no back history saved.

**Review by 5Amogh — APPROVED** [review]
